pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -151,15 +152,47 @@ type DNSProxy struct {
 	rejectReply int32
 }
 
+type FqdnsEntry struct {
+	fqdns   map[string]struct{}
+	fqdnKey *string
+}
+
+type RegexpAndFqdnsEntry struct {
+	regexp *regexp.Regexp
+	fqdns  *FqdnsEntry
+}
+
+type regexpIndexEntry struct {
+	regexp  *regexp.Regexp
+	counter int
+}
+
+type fqdnsIndexEntry struct {
+	fqdns   *FqdnsEntry
+	counter int
+}
```

Comment:
I think we can simplify some of the naming here a bit to make it easier to read. The term `Entry` is a bit overloaded here.